File: backend/src/main/java/com/example/svmarket/dto/SellerDashboardResponse.java
package com.example.svmarket.dto;

import lombok.*;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SellerDashboardResponse {
    private long totalViews;
    private int activeListingCount;
    private int soldListingCount;
    private double averageRating;
    private int reviewCount;
    private long normalViews;
    private long packageViews;
    private List<TopListingResponse> topListings;
    private List<CategoryViewResponse> categoryViews;
}

File: frontend/src/types/SellerDashboardResponse.ts
import { TopListingResponse } from "./TopListingResponse";
import { CategoryViewResponse } from "./CategoryViewResponse";

export type SellerDashboardResponse = {
    totalViews: number;
    activeListingCount: number;
    soldListingCount: number;
    averageRating: number;
    reviewCount: number;

    normalViews: number;
    packageViews: number;

    topListings: TopListingResponse[];
    categoryViews: CategoryViewResponse[];
};
